legend-engine-xts-data-push/legend-engine-xt-data-push-server/src/main/java/org/finos/legend/engine/datapush/server/ConnectionFactoryBundle.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.datapush.server;
+
+import io.dropwizard.Configuration;
+import io.dropwizard.ConfiguredBundle;
+import io.dropwizard.setup.Bootstrap;
+import io.dropwizard.setup.Environment;
+import org.finos.legend.authentication.credentialprovider.CredentialProviderProvider;
+import org.finos.legend.authentication.credentialprovider.impl.UserPasswordCredentialProvider;
+import org.finos.legend.authentication.intermediationrule.IntermediationRuleProvider;
+import org.finos.legend.authentication.intermediationrule.impl.UserPasswordFromVaultRule;
+import org.finos.legend.authentication.vault.CredentialVaultProvider;
+import org.finos.legend.authentication.vault.PlatformCredentialVaultProvider;
+import org.finos.legend.authentication.vault.impl.PropertiesFileCredentialVault;
+import org.finos.legend.connection.ConnectionFactory;
+import org.finos.legend.connection.ConnectionSetupFlowProvider;
+import org.finos.legend.connection.DefaultConnectionSetupFlowProvider;
+
+import java.util.Properties;
+import java.util.function.Function;
+
+public class ConnectionFactoryBundle<C extends Configuration> implements ConfiguredBundle<C>
```

Comment:
This should move to the base legend-engine-server-support-core module

Reply:
hm, I thought we have already put a lot of things in `legend-engine-server-support-core` but turns out right now it's pretty clean/lean, should we still chuck this bundle there? _that would introduce a fair bit of engine dependencies within `support-core`_ but effectively, that means we say any engine server core support comes with auth and connection factory capability?

Reply:
// TODO